tests/experimental/datasources/test_pandas_s3_datasource.py

```diff
@@ -190,81 +226,96 @@ def test_csv_asset_with_some_regex_named_parameters():
-    batch = batches[0]
-    assert batch.batch_request.datasource_name == pandas_s3_datasource.name
-    assert batch.batch_request.data_asset_name == asset.name
-    assert batch.batch_request.options == {
-        "path": "alex_20200819_1300.csv",
-        "name": "alex",
-        "timestamp": "20200819",
-        "price": "1300",
-    }
-    assert batch.metadata == {
-        "path": "alex_20200819_1300.csv",
-        "name": "alex",
-        "timestamp": "20200819",
-        "price": "1300",
-    }
-    assert (
-        batch.id
-        == "pandas_s3_datasource-csv_asset-name_alex-timestamp_20200819-price_1300"
-    )
+        request = asset.build_batch_request(  # type: ignore[attr-defined]
+            {"name": "alex", "timestamp": "20200819", "price": "1300"}
+        )
+        batches = asset.get_batch_list_from_batch_request(request)  # type: ignore[attr-defined]
+        assert len(batches) == 1
+        batch = batches[0]
+        assert batch.batch_request.datasource_name == pandas_s3_datasource.name
+        assert batch.batch_request.data_asset_name == asset.name  # type: ignore[attr-defined]
+        assert batch.batch_request.options == {
+            "path": "alex_20200819_1300.csv",
+            "name": "alex",
+            "timestamp": "20200819",
+            "price": "1300",
+        }
+        assert batch.metadata == {
+            "path": "alex_20200819_1300.csv",
+            "name": "alex",
+            "timestamp": "20200819",
+            "price": "1300",
+        }
+        assert (
+            batch.id
+            == "pandas_s3_datasource-csv_asset-name_alex-timestamp_20200819-price_1300"
+        )
 
-    request = asset.build_batch_request({"name": "alex"})
-    batches = asset.get_batch_list_from_batch_request(request)
-    assert len(batches) == 2
+        request = asset.build_batch_request({"name": "alex"})  # type: ignore[attr-defined]
+        batches = asset.get_batch_list_from_batch_request(request)  # type: ignore[attr-defined]
+        assert len(batches) == 2
 
 
 @pytest.mark.integration
-@mock_s3
-def test_test_connection_failures():
-    regex, test_connection_error = _build_bad_regex_config()
-    csv_asset = CSVAsset(
-        name="csv_asset",
-        batching_regex=regex,
-    )
-    pandas_s3_datasource: PandasS3Datasource = _build_pandas_s3_datasource()
-    csv_asset._datasource = pandas_s3_datasource
-    pandas_s3_datasource.assets = {"csv_asset": csv_asset}
-    csv_asset._data_connector = S3DataConnector(
-        datasource_name=pandas_s3_datasource.name,
-        data_asset_name=csv_asset.name,
-        batching_regex=re.compile(regex),
-        s3_client=_get_boto3_client(),
-        bucket=pandas_s3_datasource.bucket,
-        file_path_template_map_fn=S3Url.OBJECT_URL_TEMPLATE.format,
-    )
-    csv_asset._test_connection_error_message = test_connection_error
+def test_test_connection_failures(aws_region_name: str, aws_s3_bucket_name: str):
+    with mock_s3():
```

Comment:
Please extract this into it's own fixture. I can help with getting the context scope right.

Reply:
I already figured it out -- in next PR.  Thanks.